packages/core/src/components/axes/ruler.ts

```diff
@@ -26,8 +26,24 @@ export class Ruler extends Component {
 	}[];
 
 	render() {
+		const isRulerEnabled =  Tools.getProperty(
+			this.model.getOptions(),
+			"ruler",
+			"enabled"
+		);
+
 		this.drawBackdrop();
-		this.addBackdropEventListeners();
+
+		// remove backdrop event listeners
+		this.removeBackdropEventListeners();
+
+		if (isRulerEnabled) {
+			this.addBackdropEventListeners();
+		}
```

Comment:
so we remove & add event listeners every time? This doesn't sound healthy, there should be a better way to only remove or add when we need to

Reply:
Since d3 doesn't have an api for checking whether a specific eventListener is already existed or not. Removing and then adding event listeners when needed is what we can do for now.

Reply:
again this is the same situation as the other PR, we just need a boolean here to keep track of event listeners

Reply:
A flag, `isEventListenerAdded` has been added.